apps/agent/tests/Email/MailRouterTest.php

```
@@ -60,6 +61,446 @@ private function router(): MailRouter
+     * while the real registrations landed in this method's own (dead) local
+     * scope. Wrapping it in an object gives callers a live handle instead.
+     *
+     * @return object{filters: list<array{priority:int,callback:callable}>}
+     */
+    private function captureRegisteredPreWpMailFilters(): object
+    {
+        $box = new class () {
+            /** @var list<array{priority:int,callback:callable}> */
+            public array $filters = [];
+        };
+        Functions\when('add_filter')->alias(function (string $hook, $callback, int $priority = 10, int $acceptedArgs = 1) use ($box): bool {
```

PHPMD (2.15.0)

[warning] Avoid unused parameters such as '$acceptedArgs'. (undefined)

(UnusedFormalParameter)